### PR TITLE
[CPU] Split dynamism support

### PR DESCRIPTION
**Pull request by mandrono**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *69934*

### CODE REVIEW

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp [hunk 1]
```diff
@@ -2,79 +2,98 @@
...
+                                                                                                 netPrecision, numSplits, axis));
         ngraph::ResultVector results;
 
         for (int i = 0; i < outIndices.size(); i++) {
             // This WA is necessary because result nodes connected to the same output of the split node (or any node) are deduplicated
             // on the CNNNetwork level. It might not be needed when the CPU plugin moves completely to nGraph.
             // This is still a single layer test since the Ceiling nodes are added only as a WA.
 
-            auto fakeMultiplication = std::make_shared<ngraph::opset5::Ceiling>(split->output(outIndices[i]));
+            auto fakeMultiplication = std::make_shared<ngraph::opset5::Relu>(split->output(outIndices[i]));
```
Comment (maxnick) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/split.cpp:96]: M.b. this WA is not needed anymore?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp [hunk 2]
```diff
@@ -235,6 +230,28 @@ void MKLDNNSplitNode::initSupportedPrimitiveDescriptors() {
...
+            std::vector<BlockedMemoryDescCPtr> outDescs(outputPortsCount);
+            for (size_t i = 0; i < outputPortsCount; i++) {
+                outDescs[i] = getChildEdgesAtPort(i)[0]->getMemory().GetDescWithType<BlockedMemoryDesc>();
+            }
+            execPtr = std::make_shared<SplitOptimizedExecutor>(inDesc, outDescs, axis);
+        }
+    }
+}
+
 void MKLDNNSplitNode::createPrimitive() {
```
[the remark this answers is not in the document]
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:255]: leaved check only for `getSelectedPrimitiveDescriptor() != nullptr`
not sure that we need to check memory on nullptr in runtime

[the remark this answers is not in the document]
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:253]: restore check for mem obj

[the remark this answers is not in the document]
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:253]: need check on `nullptr` in `prepareParams` ?

**Review by maxnick — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**